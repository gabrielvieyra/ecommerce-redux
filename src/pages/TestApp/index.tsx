import { FC } from 'react';

// Styles
import './styles.scss';

interface TestAppProps {
  title: string;
  subTitle: string;
  name: string;
}

const TestApp: FC<TestAppProps> = ({ title, subTitle, name }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

export default TestApp;
