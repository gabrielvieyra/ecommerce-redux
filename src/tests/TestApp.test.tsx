import { render } from '@testing-library/react';

// Components
import TestApp from '../pages/TestApp';

describe('Pruebas en <TestApp />', () => {
  // test('Debe de hacer match con el snapshot', () => {
  // const title: string = 'Hola, Soy Goku';
  // const subTitle: string = 'No hay subtitulo';
  // const name: string = 'Gabriel Vieyra';

  // render() es una funcion que renderiza el componente en memoria
  // Este metodo retorna un objeto que expone ciertas propiedades, una de ellas es el container, este container se parece mucho a un dom que podemos utilizar para muchas cosas
  // const { container } = render(<TestApp title={title} subTitle={subTitle} name={name} />);
  // console.log(container);
  // expect(container).toMatchSnapshot();
  // });

  test('Debe de mostrar el titulo en un h1', () => {
    const title = 'Hola, Soy Goku';
    const subTitle = 'No hay subtitulo';
    const name = 'Gabriel Vieyra';
    // getByText me permite obtener un elemento por un texto
    // getByTestId me permite obtener un elemento en particular
    // toBeTruthy me permite probar que exista
    const { container, getByText, getByTestId } = render(
      <TestApp title={title} subTitle={subTitle} name={name} />
    );
    // Probamos que el componente TestApp tenga x titulo
    // expect(getByText(title)).toBeTruthy();
    expect(getByTestId('test-title').innerHTML).toBe(title);
  });
});
