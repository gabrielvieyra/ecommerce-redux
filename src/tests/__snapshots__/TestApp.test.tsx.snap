// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TestApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Gabriel Vieyra
  </p>
</div>
`;

exports[`Pruebas en <TestApp /> Debe de hacer match con el snapshot 2`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Gabriel Vieyra
  </p>
</div>
`;
